torch_xla/csrc/elementwise.cpp

```diff
@@ -227,13 +227,8 @@ xla::XlaOp BuildPrelu(xla::XlaOp input, xla::XlaOp weight) {
   const xla::Shape& input_shape = XlaHelpers::ShapeOfXlaOp(input);
   const xla::Shape& weight_shape = XlaHelpers::ShapeOfXlaOp(weight);
 
-  int64_t weight_num = xla::ShapeUtil::ElementsIn(weight_shape);
-  int64_t broadcast_dim = weight_num == 1 ? 0 : 1;
-
   xla::XlaOp zero = xla::Zero(input.builder(), input_shape.element_type());
-  xla::XlaOp broadcasted_weight =
-      xla::BroadcastInDim(weight, input_shape.dimensions(), {broadcast_dim});
-  xla::XlaOp product = xla::Mul(input, broadcasted_weight);
+  xla::XlaOp product = xla::Mul(input, weight);
```

Comment:
Can you explain a little bit on this change?

Reply:
Now, the `at::Tensor& weight` that is passed to `XLANativeFunctions::_prelu_kernel` already broadcasted. So updated our lowering to remove the unnecessary broadcasting logic. 

Also updated in the PR summary above to include this reasoning.

Reply:
Sounds good to me.